include/dxc/DXIL/DxilResource.h

```diff
@@ -29,8 +29,19 @@ class DxilResource : public DxilResourceBase {
   static unsigned GetNumDimensionsForCalcLOD(Kind ResourceKind);
   /// Total number of offsets (in [-8,7]) necessary to access resource.
   static unsigned GetNumOffsets(Kind ResourceKind);
-  /// Whether the resource kind is texture.
+  /// Whether the resource kind is a texture. This does not include
+  /// FeedbackTexture.
   static bool IsAnyTexture(Kind ResourceKind);
+  /// Whether the resource kind is an array of textures. This does not include
+  /// texture feedback arrays.
+  static bool IsAnyTextureArray(Kind ResourceKind);
+  /// Whether the resource kind is a texture cube or an array of texture cubes.
+  static bool IsAnyTextureCube(Kind ResourceKind);
+  /// Whether the resource kind is a FeedbackTexture, or an array of
+  /// FeedbackTexture.
```

Comment:
I guess this still bothers me.  Either the method is self-explanatory in this case (I think it is), or you want to list the actual types it matches, which is FeedbackTexture2D and FeedbackTexture2DArray, but not "an array of FeedbackTexture".  A Texture\*Array isn't an array *of* textures, but instead a single texture resource view containing an additional array dimension along with the other dimensions (1D, 2D, or Cube) within the same resource allocation.  An array of textures from the HLSL point of view is an array of resource view descriptors (SRV/UAV) that each point to a different resource view with the same kind.  I think you could use "FeedbackTextures" to indicate any of the FeedbackTexture\* types (this applies to the prior comments as well).

This comment also applies to other comments, like for `IsAnyArray()`.  In fact, given this confusion, and given that a single `DxilResource` may in fact represent *an array of resources*, `IsAnyArray()` is actually a bit ambiguous, particularly the non-static method below, since it doesn't refer to whether the `DxilResource` represents an array of resources, but whether the kind is a Texture or FeedbackTexture with an Array dimension.  Maybe instead, `HasArrayDimension()` would be more clear.  Plus `IsAnyTextureArray()` still seems extra, since you could use `IsAnyTexture() && HasArrayDimension()` instead, and it still has that ambiguity problem (array of textures or Texture*Array type?).  I'm not sure how to resolve the ambiguity problem without giving it an uglier name, such as `IsAnyTextureWithArrayDimension()`, or just removing it.

Reply:
Or maybe `KindHasArrayDimension()` or `IsArrayResource()` or `IsArrayResourceKind()` or `IsArrayKind()` instead of `HasArrayDimension()` depending on what seems most concise and clear.  I kind of like that last one: `IsArrayKind()`.